java/integTest/src/test/java/glide/SharedCommandTests.java

```diff
@@ -1168,4 +1169,58 @@ public void type(BaseClient client) {
         assertTrue("zset".equalsIgnoreCase(client.type(zsetKey).get()));
         assertTrue("stream".equalsIgnoreCase(client.type(streamKey).get()));
     }
+
+    @SneakyThrows
+    @ParameterizedTest
+    @MethodSource("getClients")
+    public void blpop(BaseClient client) {
+        String listKey1 = "{listKey}-1-" + UUID.randomUUID();
+        String listKey2 = "{listKey}-2-" + UUID.randomUUID();
+        String value1 = "value1-" + UUID.randomUUID();
+        String value2 = "value2-" + UUID.randomUUID();
+        assertEquals(2, client.lpush(listKey1, new String[] {value1, value2}).get());
+
+        var response = client.blpop(new String[] {listKey1, listKey2}, 0.5).get();
+        assertArrayEquals(new String[] {listKey1, value2}, response);
+
+        // nothing popped out
+        assertNull(
+                client
+                        .blpop(new String[] {listKey2}, REDIS_VERSION.isLowerThan("7.0.0") ? 1. : 0.001)
```

Comment:
What is that you are testing via the ternary operator?

Reply:
Redis 6 does not support FP values. It rounds 0.0001 to 0 and blocks infinitely.